### DISCUSSION TITLE
Question for workflow after sign up - store additional information

### DISCUSSION THREAD

**Discussion by megacherry** [supabase/supabase#1751] [Questions]
Hi at all,

first thing: I switched from firebase to supabase and it's awesome. But i have a question about the sign up process. I have an application where you have to register an account. You can choose your email, username and password. In the supabase-backend i have a trigger on auth.users to copy the user-information to public.profiles. In this table i have a column called username.

Do you have an approach or a simple idea how i can also update the username at the sign up process? On the profiles-table there is also a policy which prevent to update data if auth.uid() = id.

I would prefer a solution in the backed, for example a new parameter for the signUp function.

Thanks for tipps and tricks :)

Best regards
Andreas

**Comment by quicksnap** [comment]
In my app, I have the concept of "profile initializing". I allow them to sign up, but then force them to insert into a public `user_profiles` table before they may do anything. I am likely going to create another security definer function called `auth.initializedUid()` for row policies to ensure only users that have finished the initial setup may continue.

Curious if there are other approaches.

**Reply by GrbavaCigla** [reply]
I have the same problem, but I cannot use this solution because `username` column is required for everything on the backend. If I do that, I would have to create policy for every table. Is there any better solution?

**Reply by quicksnap** [reply]
I'm not sure I follow. Could you elaborate? What do you mean "`username` column is required for everything.."?

**Reply by GrbavaCigla** [reply]
I want to have username just as a displayname. If a user creates a post, username cannot be empty, so username is required to create a post.

**Reply by quicksnap** [reply]
I'm still not clear on the actual problem at hand, though. Sorry.. not trying to be obstinate or anything, just not seeing where you're running into trouble.

**Reply by GrbavaCigla** [reply]
Basically I want to have a RLS policy where I check if user creating a post has username.

**Reply by GrbavaCigla** [reply]
I found acceptable solution:
```sql
(
  (uid() = author_id)
  AND (
    (
      SELECT
        profiles.username
      FROM
        profiles
      WHERE
        (profiles.id = uid())
      LIMIT
        1
    ) IS NOT NULL
  )
)
```
Thanks for your help :D
